kube-core/src/request.rs

```diff
@@ -754,24 +772,23 @@ mod test {
     #[test]
     fn list_invalid_resource_version_combination() {
         let url = corev1::Pod::url_path(&(), Some("ns"));
-        let gp = ListParams::default().resource_version_match(ResourceVersionMatch::NotOlderThan);
+        let gp = ListParams::default().version_match(VersionMatch::NotOlderThan("0".to_string()));
         let err = Request::new(url).list(&gp).unwrap_err();
         assert!(format!("{err}")
-            .contains("resource_version cannot be empty if the resource_version_match is set"));
+            .contains("version_match cannot be equal to \"0\" for Exact and NotOlderThan variants"));
     }
 
     #[test]
-    fn list_invalid_resource_version_zero_and_exact_match() {
+    fn list_exact_match() {
         let url = corev1::Pod::url_path(&(), Some("ns"));
-        let gp = ListParams::default()
-            .resource_version("0")
-            .resource_version_match(ResourceVersionMatch::Exact);
-        let err = Request::new(url).list(&gp).unwrap_err();
-        assert!(format!("{err}")
-            .contains("resource_version cannot be equal to \"0\" if the resource_version_match is Exact"));
+        let gp = ListParams::default().version_match(VersionMatch::Exact("500".to_string()));
+        let req = Request::new(url).list(&gp).unwrap();
+        assert_eq!(
+            req.uri(),
+            "/api/v1/namespaces/ns/pods?&resourceVersion=500&resourceVersionMatch=Exact"
```

Comment:
would be nice to have similar test cases for the other enum variants here as well. can also slim them down to 3 lines by only testing the thing we actually care about; the query string:

```rust
let gp2 = ListParams::default().version_match(VersionMatch::NotOlderThan("20".into()));
let uri2 = Request::new(url).list(&gp2).unwrap().uri();
assert_eq!(uri2.query(), "&resourceVersion=20&resourceVersionMatch=NotOlderThan");
```

Reply:
Added more tests.